src/main/java/core/basesyntax/service/impl/ReturnTransactionHandler.java

```
@@ -0,0 +1,16 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Database;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.service.TransactionHandler;
+import java.math.BigInteger;
+
+public class ReturnTransactionHandler implements TransactionHandler {
+    @Override
+    public void handle(Database database, FruitType fruitType, BigInteger quantity) {
+        if (quantity == null || quantity.signum() < 0) {
+            throw new IllegalArgumentException("Quantity is wrong. Cannot be: " + quantity);
+        }
+        database.add(fruitType, quantity);
```

After validation the handler calls `database.add(fruitType, quantity)` without ensuring the database method will handle nulls or unexpected keys. Either validate `fruitType` before calling database.add (see first comment) or document/ensure that Database implementations validate their inputs. Failing fast in this handler with an informative message is preferable to letting a NullPointerException occur deeper in the stack. Consider adding a comment about expected Database behavior or adding defensive checks here.